### PR TITLE
CedMiddleware documentation [4633]

### PR DESCRIPTION
**Pull request by julianbermudez**
This pull request adds the CedMiddleware documentation using Doxygen comments.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Agent%20Manual%20Linux/157/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Agent%20Manual%20Windows/160/)

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp [hunk 1]
```diff
@@ -28,123 +28,339 @@ class CedMiddleware : public Middleware
...
+     * !include agent/middleware/ced/create_participant.puml!0
+     * @enduml
+     *
+     * @brief Creates a CedParticipant from a reference.
+     *        Currently, only the domain_id is taken into account for the creation.
+     *        In a near future, the Middleware interface should change the ref parameter by an "attributes" one.
+     *        For this purpose an XML parser and a reference data base are needed in the XRCE Core.
+     * @param participant_id    The CedParticipant identifier.
+     * @param domain_id         The Domain identifier to which the CedParticipant belongs.
+     * @param ref               The CedParticipant reference (not used).
```
Comment (BorjaOuterelo) [inline include/uxr/agent/middleware/ced/CedMiddleware.hpp]: I would do (not used) more outstanding, `NOT USED`?

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp [hunk 2]
```diff
@@ -28,123 +28,339 @@ class CedMiddleware : public Middleware
...
+     * @startuml
+     * !include agent/middleware/ced/create_publisher.puml!0
+     * @enduml
+     *
+     * @brief Creates a CedPublisher associated to a CedParticipant.
+     * @param publisher_id      The CedPublisher identifier.
+     * @param participant_id    The CedParticipant identifier to which the CedPublisher is associated.
+     * @return  true in case of creation and false in other case.
+     */
     bool create_publisher_by_xml(
```
Comment (BorjaOuterelo) [inline include/uxr/agent/middleware/ced/CedMiddleware.hpp]: There is not `by_ref` version?

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp [hunk 3]
```diff
@@ -28,123 +28,339 @@ class CedMiddleware : public Middleware
...
+     * @startuml
+     * !include agent/middleware/ced/create_subscriber.puml!0
+     * @enduml
+     *
+     * @brief Creates a CedSubscriber associated to a CedParticipant.
+     * @param subscirber_id     The CedSubscriber identifier.
+     * @param participant_id    The CedParticipant identifier to which the CedSubscriber is associated.
+     * @return  true in case of creation and false in other case.
+     */
     bool create_subscriber_by_xml(
```
Comment (BorjaOuterelo) [inline include/uxr/agent/middleware/ced/CedMiddleware.hpp]: There is not `by_ref` version?

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp [hunk 4]
```diff
@@ -28,123 +28,339 @@ class CedMiddleware : public Middleware
...
- **********************************************************************************************************************/
+    /**
+     * @brief Removes a CedParticipant from the participants register.
+     * @param participant_id    The CedParticipant identifier.
+     * @return  true in case of the CedParticipant was found and removed, false in other case.
+     */
     bool delete_participant(uint16_t participant_id) override;
 
+    /**
+     * @brief Removes a CedParticipant from the topic register.
```
Comment (BorjaOuterelo) [inline include/uxr/agent/middleware/ced/CedMiddleware.hpp]:
```suggestion
     * @brief Removes a CedTopic from the topic register.
```

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp [hunk 5]
```diff
@@ -28,123 +28,339 @@ class CedMiddleware : public Middleware
...
 
+    /**
+     * @brief Removes a CedSubscriber from the subscriber register.
+     * @param subscriber_id The CedSubscriber identifier.
+     * @return  true in case of the CedSubscriber was found and removed, false in other case.
+     */
     bool delete_subscriber(uint16_t subscriber_id) override;
 
+    /**
+     * @brief Removes a DataWriter from the datawriters register.
```
Comment (BorjaOuterelo) [inline include/uxr/agent/middleware/ced/CedMiddleware.hpp]:
```suggestion
     * @brief Removes a CedDataWriter from the datawriters register.
```

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp [hunk 6]
```diff
@@ -28,123 +28,339 @@ class CedMiddleware : public Middleware
...
+     * @param datawriter_id The CedDataWriter identifier.
+     * @return  true in case of the CedDataWriter was found and removed, false in other case.
+     */
     bool delete_datawriter(uint16_t datawriter_id) override;
 
+    /**
+     * @brief Removes a DataReader from the datawriters register.
```
Comment (BorjaOuterelo) [inline include/uxr/agent/middleware/ced/CedMiddleware.hpp]:
```suggestion
     * @brief Removes a CedDataReader from the datawriters register.
```

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp [hunk 7]
```diff
@@ -28,123 +28,339 @@ class CedMiddleware : public Middleware
...
- **********************************************************************************************************************/
+    /**
+     * @brief Checks whether an existing CedParticipant, identified by the participant_id, matches with a new
+     *        CedParticipant that would result from the creation of a new one using the domain_id and the reference
+     *        representation.
+     *        It is considered that the CedParticipants match if both have the same domain_id.
+     * @param participant_id    The existing CedParticipant identifier.
+     * @param domain_id         The Domain identifier of the new CedParticipant.
+     * @param ref               The reference that represents the new CedParticipant (unused).
+     * @return  true if both CedParticipant have the same Domain identifier, and false in other case.
```
Comment (BorjaOuterelo) [inline include/uxr/agent/middleware/ced/CedMiddleware.hpp]: This is true?
Do two participants match just matching the Domain?.

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp [hunk 8]
```diff
@@ -28,123 +28,339 @@ class CedMiddleware : public Middleware
...
 
+    /**
+     * @brief Checks whether an existing CedParticipant, identified by the participant_id, matches with a new
+     *        CedParticipant that would result from the creation of a new one using the domain_id and the XML
+     *        representation.
+     *        It is considered that the CedParticipants match if both have the same domain_id.
+     * @param participant_id    The existing CedParticipant identifier.
+     * @param domain_id         The Domain identifier of the new CedParticipant.
+     * @param xml               The XML that describes the new CedParticipant (unused).
+     * @return  true if both CedParticipant have the same Domain identifier, and false in other case.
```
Comment (BorjaOuterelo) [inline include/uxr/agent/middleware/ced/CedMiddleware.hpp]: Same.

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp [hunk 9]
```diff
@@ -28,123 +28,339 @@ class CedMiddleware : public Middleware
...
             const std::string& xml) const override;
 
+    /**
+     * @brief Checks whether an existing CedTopic, identified by the topic_id, matches with a new CedTopic that would
+     *        result from the creation of a new one using the reference representation.
+     *        It is considered that the CedTopics match if both have the same topic name.
+     * @param topic_id  The existing CedTopic identifier.
+     * @param ref       The reference that represents the new CedTopic.
+     *                  It defines the CedGlobalTopic name.
+     * @return  true if both CedTopic have associated the same CedGlobalTopic, and false in other case.
```
Comment (BorjaOuterelo) [inline include/uxr/agent/middleware/ced/CedMiddleware.hpp]: What is CedGlobalTopic?.

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp [hunk 10]
```diff
@@ -28,123 +28,339 @@ class CedMiddleware : public Middleware
...
+     * @return true if both CedDataWriter have associated the same CedGlobalTopic name
+     */
     bool matched_datawriter_from_ref(
             uint16_t datawriter_id,
             const std::string& ref) const override;
 
+    /**
+     * @brief Checks whether an existing CedDataWriter, identified by the datawriter_id, matches with the new
+     *        CedDataWriter that would result from the creation of a new one using the XML representation.
+     *        It is considered that the CedDataWriters match if both are associated to the same CedGlobalTopic.
```
Comment (BorjaOuterelo) [inline include/uxr/agent/middleware/ced/CedMiddleware.hpp]: Are they just matching `CedGlobalTopic`? Do not `datawriter_id` participate in the matching?

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp (hunk 2 above)
Comment (julianbermudez) [reply include/uxr/agent/middleware/ced/CedMiddleware.hpp]: No, it is not in the standard :(

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp (hunk 3 above)
Comment (julianbermudez) [reply include/uxr/agent/middleware/ced/CedMiddleware.hpp]: No, it is not in the standard :(

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp (hunk 7 above)
Comment (julianbermudez) [reply include/uxr/agent/middleware/ced/CedMiddleware.hpp]: Yes, in CedMiddleware it is true.

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp (hunk 8 above)
Comment (julianbermudez) [reply include/uxr/agent/middleware/ced/CedMiddleware.hpp]: Same.

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp (hunk 9 above)
Comment (julianbermudez) [reply include/uxr/agent/middleware/ced/CedMiddleware.hpp]: It is the Topic which is shared between all the Participants, DataWriter and DataReader.

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp (hunk 10 above)
Comment (julianbermudez) [reply include/uxr/agent/middleware/ced/CedMiddleware.hpp]: No, just the `CedGlobalTopic`.

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp (hunk 1 above)
Comment (julianbermudez) [reply include/uxr/agent/middleware/ced/CedMiddleware.hpp]: Fixed!

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp (hunk 4 above)
Comment (julianbermudez) [reply include/uxr/agent/middleware/ced/CedMiddleware.hpp]: Fixed!

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp (hunk 5 above)
Comment (julianbermudez) [reply include/uxr/agent/middleware/ced/CedMiddleware.hpp]: Fixed!

File: include/uxr/agent/middleware/ced/CedMiddleware.hpp (hunk 6 above)
Comment (julianbermudez) [reply include/uxr/agent/middleware/ced/CedMiddleware.hpp]: Fixed!

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Agent%20Manual%20Linux/204/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Agent%20Manual%20Windows/207/)

**Pull request merged**